csv-data-management/extract_data_from_csv.py

```diff
@@ -280,8 +280,11 @@ def _check_if_csv(filepath):
     # To-Do  Read CSV file encoding and then use it for reading file. use -chardet.detect
     try:
         res = sniffer.has_header(open(filepath).read(2048))
-    except UnicodeDecodeError:
-        raise ConnectorError("CSV file has unsupported encoding.Supported encoding is UTF-8")
+    except Exception as Err:
+        if "UnicodeDecodeError" in repr(Err):
+            raise ConnectorError("CSV file has unsupported encoding.Supported encoding is UTF-8")
+        else:
+            pass
```

Comment:
why pass?

Reply:
as i am checking only for UnicodeDecodeError and the bailing out. If UnicodeDecodeError is not present I re-read file to check for headers. those exceptions are handled in below